options/posix/generic/unistd-stubs.cpp

```diff
@@ -123,9 +123,24 @@ int execle(const char *path, const char *arg0, ...) {
 	return execve(path, argv, envp);
 }
 
-int execlp(const char *, const char *, ...) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+// This function is taken from musl
+int execlp(const char *file, const char *argv0, ...) {
+	int argc;
+	va_list ap;
+	va_start(ap, argv0);
+	for(argc = 1; va_arg(ap, const char *); argc++);
+	va_end(ap);
+	{
+		int i;
+		char *argv[argc + 1];
```

Comment:
I do not like the use of VLA's here. I think it's fine here to malloc an array and never free it, since ``argv`` for a program is never freed anyways.

Reply:
Considering it's already in mlibc master (see [here](https://github.com/managarm/mlibc/blob/bc8dd2e192018ddb0c49be49a4885ddd978f4028/options/posix/generic/unistd-stubs.cpp#L115)), I'd say this is fine for now.

Reply:
I agree, we should open a separate issue for use of VLAs.